packages/ListBoxWithTags/src/useListBoxWithTags.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import { filter } from "./helpers";
+
+function prepareDataDictionary(key, data) {
+  const dictionary = {};
+
+  data.forEach(d => {
+    dictionary[d[key]] = d;
+  });
+
+  return dictionary;
+}
+
+export default function useListBoxWithTags(
+  key,
+  { defaultData = [], defaultFilteredData = [], defaultSelectedKeys = [], filterAttribute = "label" }
+) {
+  if (!key)
+    throw new Error("Key is Required for useListBoxWithTags and should be one of the key (string) in your object");
+
+  const [data, setData] = React.useState(defaultData);
+  const { t } = useI18n();
+  const [selectedKeys, setSelectedKeys] = React.useState(defaultSelectedKeys);
+  const [filteredData, setFilteredData] = React.useState(defaultFilteredData);
+
+  const dataDictionary = React.useMemo(() => {
+    return prepareDataDictionary(key, data);
+  }, [data, key]);
+
+  function handleChange(option, options, selectedOption) {
+    setSelectedKeys(prev => {
+      const prevClone = prev.slice(0);
+      const id = options[selectedOption].content.props[key];
+      if (prev.includes(id)) {
+        prev.splice(prevClone.indexOf(id), 1);
+        return prev;
+      }
+
+      return [...new Set(prevClone.concat([id]))];
+    });
+
+    setFilteredData(defaultData);
+  }
+
+  function getSelectedOptions() {
+    if (selectedKeys.length) {
+      const options = [];
```

Comment:
Put `options` outside of the `if` statement, and at the end of this function `return options;`